File: app/src/main/java/com/vpulse/ftpnext/core/FTPCharacterEncoding.java
package com.vpulse.ftpnext.core;

public enum FTPCharacterEncoding {

    UTF_8(0),

    DEFAULT(0);

    private final int mValue;

    FTPCharacterEncoding(int iValue) {
        this.mValue = iValue;
    }

    public int getValue() {
        return mValue;
    }

    public static FTPCharacterEncoding getValue(int iValue) {
        if (iValue == 1)
            return FTPCharacterEncoding.UTF_8;
        return FTPCharacterEncoding.DEFAULT;
    }
}

File: app/src/main/java/com/vpulse/ftpnext/core/FTPCharacterEncoding.kt
package com.vpulse.ftpnext.core

enum class FTPCharacterEncoding(val value: Int) {
    UTF_8(0), DEFAULT(0);

    companion object {
        fun getValue(iValue: Int): FTPCharacterEncoding {
            return if (iValue == 1) UTF_8 else DEFAULT
        }
    }
}